File: solutions/r/flatten-array/1/flatten-array.R
flatten <- function(input) {
  result <- c()

  for (item in input) {
    if (is.null(item)) {
      next
    }

    if (is.list(item)) {
      result <- c(result, flatten(item))
    } else {
      result <- c(result, item)
    }
  }

  result
}

File: solutions/vbnet/flatten-array/1/FlattenArray.vb
Imports System
Imports System.Collections

Public Module FlattenArray
    Public Iterator Function Flatten(input As IEnumerable) As IEnumerable
        For Each i As Object In input
            If TypeOf i Is IEnumerable Then
                For Each v As Object In Flatten(DirectCast(i, IEnumerable))
                    Yield v
                Next
            ElseIf i IsNot Nothing Then
                Yield i
            End If
        Next
    End Function
End Module